File: 0001-0500/0035/0035.search-insert-position.cpp
/*
 * @lc app=leetcode id=35 lang=cpp
 *
 * [35] Search Insert Position
 */

// @lc code=start
class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int l = nums.size() - 1;
        int pos = 0;
        while (l >= pos)
        {
            int mid = (l + pos) / 2;
            if (nums[mid] < target)
            {
                pos = mid + 1;
            }
            if (nums[mid] == target)
            {
                return mid;
            }
            if (nums[mid] > target)
            {
                l = mid - 1;
            }
        }
        return pos;
    }
};
// @lc code=end

File: 0001-0500/0035/0035.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python3
#
# [35] Search Insert Position
#

# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums)-1
        while start <= end:
            mid = (start + end)//2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
        return end+1
# @lc code=end

